tensorflow/compiler/xla/service/hlo_verifier.cc

```diff
@@ -406,10 +406,20 @@ static Status CheckCommonAllGatherInvariants(HloInstruction* hlo,
                                                ag->use_global_device_ids()));
   TF_RETURN_IF_ERROR(CheckReplicaGroups(ag, group_mode));
   TF_RET_CHECK(ag->all_gather_dimension() >= 0);
+  TF_RET_CHECK(ag->operand_count() >= 1);
 
   int64_t shard_count;
+  // There can be one token in the input Tuple. The token is a scalar or
```

Comment:
This is confusing IMO. Can we get clarification as to what is intent of the token input to all-gather? Also treating a scalar as a token is confusing.